src/cyclonedds/cyclonedds/core.py

```diff
@@ -1510,6 +1512,13 @@ def set_trigger(self, value: bool) -> None:
         if ret < 0:
             raise DDSException(ret, f"Occurred when setting trigger in {repr(self)}")
 
+    async def wait_async(self, timeout: Optional[int] = None) -> int:
+        timeout = timeout or 0xFFFFFFFFFFFFFFFF
```

Comment:
The `0xFFF...`, where does that come from? Does that mean infinity/no timeout? There's probably some constant that's better used in this case.

Reply:
Ah yes I had planned to implement and use `cyclonedds.util.duration(infinite=True)` but this would cause a circular import. I will add `dds_infinity` to `internal.py` and implement that both in `duration` and here in the waitset.